src/services/pull/usePullBundleList.spec.tsx

```diff
@@ -0,0 +1,297 @@
+  owner: {
+    repository: {
+      __typename: 'OwnerNotActivatedError',
+      message: 'owner not activated',
+    },
+  },
+}
+
+const server = setupServer()
+const queryClient = new QueryClient({
+  defaultOptions: {
+    queries: {
+      retry: false,
+    },
+  },
+})
+
+const wrapper: React.FC<React.PropsWithChildren> = ({ children }) => (
+  <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
+)
+
+beforeAll(() => {
+  server.listen()
+})
+
+afterEach(() => {
+  queryClient.clear()
+  server.resetHandlers()
+})
+
+afterAll(() => {
+  server.close()
+})
+
+interface SetupArgs {
+  isUnsuccessfulParseError?: boolean
+  isNullOwner?: boolean
+  isOwnerNotActivatedError?: boolean
+  isNotFoundError?: boolean
+}
+
+describe('usePullBundleList', () => {
+  function setup({
+    isUnsuccessfulParseError = false,
+    isNullOwner = false,
+    isNotFoundError = false,
+    isOwnerNotActivatedError = false,
+  }: SetupArgs = {}) {
+    server.use(
+      graphql.query('PullBundleList', (req, res, ctx) => {
+        if (isNotFoundError) {
+          return res(ctx.status(200), ctx.data(mockNotFoundError))
+        } else if (isOwnerNotActivatedError) {
+          return res(ctx.status(200), ctx.data(mockOwnerNotActivatedError))
+        } else if (isUnsuccessfulParseError) {
+          return res(ctx.status(200), ctx.data(mockUnsuccessfulParseError))
+        } else if (isNullOwner) {
+          return res(ctx.status(200), ctx.data(mockNullOwner))
+        } else {
+          return res(ctx.status(200), ctx.data(mockPullBundleListData))
+        }
+      })
+    )
+  }
+
+  describe('api returns valid response', () => {
+    it('returns pull summary data', async () => {
+      setup()
+      const { result } = renderHook(
+        () =>
+          usePullBundleList({
+            provider: 'github',
+            owner: 'codecov',
+            repo: 'test-repo',
+            pullId: 123,
+          }),
+        { wrapper }
+      )
+
+      const expectedResult = {
+        owner: {
+          repository: {
+            __typename: 'Repository',
+            commit: {
+              bundleAnalysisCompareWithParent: {
+                __typename: 'BundleAnalysisComparison',
+                bundles: [
+                  {
+                    name: 'bundle.js',
+                    changeType: 'added',
+                    sizeDelta: 1,
+                    sizeTotal: 2,
+                    loadTimeDelta: 3,
+                    loadTimeTotal: 4,
+                  },
+                  {
+                    name: 'bundle.css',
+                    changeType: 'removed',
+                    sizeDelta: 5,
+                    sizeTotal: 6,
+                    loadTimeDelta: 7,
+                    loadTimeTotal: 8,
+                  },
+                ],
+              },
+            },
+          },
+        },
+      }
+
+      await waitFor(() =>
+        expect(result.current.data).toStrictEqual(expectedResult)
```

Comment:
Seems like this is happening because of failing to parse zod schema so you might have to change something there.